src/movies.js

```diff
@@ -1,16 +1,149 @@
 // Iteration 1: All directors? - Get the array of all directors.
+function getAllDirectors(arr) {
+    let newArr = arr.map(function(elem){
+        return elem.director;
+    })
+    return newArr;
+}
+
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
 
+function howManyMovies(arr) {
+    let spielbergArr = arr.filter(function(elem){
+        return elem.director == "Steven Spielberg" && elem.genre.includes('Drama');
+    })
+    return spielbergArr.length;
+}
+
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
 
+function ratesAverage(arr) {
+    let rateArray = arr.filter(function(elem){
+        return elem.hasOwnProperty('rate') == true;
+    });
+
+    let avgRate = rateArray.reduce(function(acc, elem){
+        
+        return acc + elem.rate/arr.length;
+    }, 0);
+
+
+    return Math.round((avgRate + Number.EPSILON) * 100) / 100;
+}
+
 // Iteration 4: Drama movies - Get the average of Drama Movies
 
+function dramaMoviesRate(arr) {
+    let dramaArray = arr.filter(function(elem){
+        return elem.genre.includes('Drama');
+    });
+
+    let avgRate = dramaArray.reduce(function(acc, elem){
+        
+        return acc + elem.rate/dramaArray.length;
+    }, 0);
+
+
+    return Math.round((avgRate + Number.EPSILON) * 100) / 100;
+}
+
+
 // Iteration 5: Ordering by year - Order by year, ascending (in growing order)
 
+function orderByYear(arr) {
+    let chronologArr = JSON.parse(JSON.stringify(arr));
+    chronologArr.sort(function(a, b){
+      if (a.year > b.year) {
+        return 1;
+      }
+      else if (a.year < b.year) {
+        return -1;
+      }
+      else {
+        if (a.title > b.title) {
+            return 1;
+          }
+          else if (a.title < b.title) {
+            return -1;
+          }
+          else {
+            return 0;
+          }
+      }
+    })
+    return chronologArr;
+  }
+
 // Iteration 6: Alphabetic Order - Order by title and print the first 20 titles
 
+function orderAlphabetically(arr) {
+    let alphabArray = JSON.parse(JSON.stringify(arr));
+    alphabArray.sort(function(a, b){
+        if (a.title > b.title) {
+            return 1;
+        }
+        if (a.title < b.title) {
+            return -1;
+        }
+        else {
+            return 0;
+        }
+    })
+
+    let titleArray = alphabArray.map(function(elem){
+        return elem.title;
+    })
+    return titleArray.slice(0, 20);
+}    
+
 // BONUS - Iteration 7: Time Format - Turn duration of the movies from hours to minutes
 
+
+function turnHoursToMinutes(arr) {
+
```

Comment:
Even though you are using .map to create a new array you are still accessing objects via the parameter elem. And because those point to the original objects you are mutating them, so either you will need to make a copy of the array before the map or make a copy of the object inside the map.